### ISSUE TITLE
pi: download bandwidth is inaccurate

### ISSUE THREAD

**Issue by scruzin** [ausocean/client#1]
The X1 pin reports download bandwidth in bits per second (bps), which is computed by timing how long it takes to download a file from the cloud service (VidGrind or DataBlue). It is computing ridiculously large values. The following is the relevant code snippet.

```
	now := time.Now()
	resp, err := http.Get(url)
	if err != nil {
		return fmt.Errorf("could not do download speed test request: %w", err)
	}
	dur := time.Now().Sub(now).Seconds()

	// Calculate download speed in bits/s.
	ns.download = int((downloadTestSize * 8) / dur)
```

We use the same methodology to compute the upload time, which is reasonably accurate. The notable difference is that uploads take much longer.

Hitting DataBlue manually (using curl), produces a realistic time:
```
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 1220k  100 1220k    0     0  1055k      0  0:00:01  0:00:01 --:--:-- 1055k

real    0m1.234s
user    0m0.172s
sys     0m0.167s
```
125000 bytes (or 1 Megabit) in 0.167s equals 59.8 Mbps, which is accurate.

I'm speculating that time.Now() is inaccurate on Raspberry Pi and is resulting in a smaller duration (dur) than actually elapsed.

**Comment by scruzin** [comment]
Actually, I think the client code is flawed. Calling http.Get does not automatically read the response body at the network level.  So we're only timing how long it takes to perform the GET, not the time to read the body.

**Comment by scruzin** [comment]
Implemented in Pull Request #2
